content/6_Advanced/Game_Theory.mdx

```diff
@@ -40,10 +38,285 @@ frequency: 0
     />
 </Resources>
 
-<br />
+## NIM Game
+
+<FocusProblem problem="nim" />
+
+### Game description
+
+The most well-known impartial game is the NIM game. In this game there are considered $n$ piles with stones. At every step, a player removes a non-zero number of stone from a pile.
+The game continues while there are stones left. The winner is the player that makes the last move.
+
+Let's denote a state of the game as ${x_1, x_2, x_3, \ldots, x_n}$, where $x_i$ is the number of stones left in pile $i$. Then the winning strategy is as following:
+$$
+\text{The current player has a winning strategy if and only if the xor0sum of the pile sizes is not } 0
+$$
+
+The xor sum is $x_1 \oplus x_2 \oplus x_3 \ldots \oplus x_n$, where $\oplus$ denotes the [bitwise exclusive xor](/adv/xor-basis).
+
+<Info title="NIM game with increases">
+In some cases there could be allowed to also **add** stones to a chosen pile, so when a turn comes, you can remove or add stones. It turns out that adding doesn't change the how winning and losing states are determined i.e. adding operation doesn't affect the outcome of the game.
+</Info>
+
+
+### Demonstration
+
+To demonstrate this theory we have to show that from a state with xor-sum $0$ (losing), whatever the move is, we can only reach a state with non-zero xox-sum (winning), and from a winning state we can reach a losing state.
+
+We will assume that from a state with sum-xor $0$ we can also reach another state with xor-sum $0$. Let's pick an arbitrary pile with $y$ stones and we would like to remove $x$ stones ($x \in [1, y]$).
```

Comment:
why is one sum-xor and the other xor-sum

Reply:
Fixed